docs/enhancements/hypershift-adaptation.md

```diff
@@ -0,0 +1,99 @@
+# Proposal: Performance Profile Controller Adaptation to Hypershift
+
+## Additional information
+
+### Hypershift docs
+
+Documentation worth reading to get familiar with Hypershift.
+
+- High level approach : <https://hypershift-docs.netlify.app/>
+- Main concepts and actors: <https://hypershift-docs.netlify.app/reference/concepts-and-personas/>
+- Architecture: <https://hypershift-docs.netlify.app/reference/controller-architecture/>
+
+## Context
+
+Performance Profile Controller optimizes OpenShift clusters for applications sensitive to cpu and network.
+More info [here](https://github.com/openshift/cluster-node-tuning-operator/blob/master/docs/performanceprofile/performance_controller.md)
+
+[PerformanceProfile CRD](https://github.com/openshift/cluster-node-tuning-operator/blob/master/docs/performanceprofile/performance_profile.md) is the API for Performance Profile Controller. Using a PerformanceProfile as input Performance Profile Controller currently generates four different objects:
+
+- machineconfiguration.openshift.io/v1/MachineConfiguration
+- machineconfiguration.openshift.io/v1/KubeletConfig
+- Tuned
+- RuntimeClass
+
+In a Standalone deployment this CRs are generated by Performance Profile Controller and then reconciled by the appropriate controllers/operators running in the same cluster.
+In a Hypershift deployment we need to change how these CRs are created.
+
+__Note__: As Performance Profile Controller is part of [Cluster Node Tuning Operator](https://github.com/openshift/cluster-node-tuning-operator) (NTO) the proposal below will reference some of the NTO adaptations to Hypershift.
+
+Here are the documents as reference:
+
+- [Enhancement](https://github.com/openshift/enhancements/pull/1229)
+- [NTO Phase one adaptation](https://github.com/openshift/cluster-node-tuning-operator/pull/390)
+- [Hypershift changes for NTO adaptation](https://github.com/openshift/hypershift/pull/1651)
+- [NTO Phase two adaptation proposal](https://github.com/openshift/hypershift/pull/1729)
+- [NTO Hypershift adaptation design document](https://docs.google.com/document/d/1G9uu_EBCu-X8OgEA5L0P1GZMcrASjQJ6qIYAJSS6NbY/edit)
+
+## Proposal
+
+### Performance Profile
+
+The proposal to handle the Performance Profile (PP) mimics the way NTO handles it `Tuned` API
+
+- New `.spec.performanceProfileConfig` will be created in `NodePool` CRD
+- [Cluster Service Consumer](https://hypershift-docs.netlify.app/reference/concepts-and-personas/) will create `PerformanceProfile` objects inside of configmaps, then reference these configmaps objects in the `.spec.performanceProfileConfig` field of the `NodePool` API
```

Comment:
This will depend on what the HyperShift team prefers, but I wonder if it would be better to use the same `.spec.tunedConfig` field for referencing ConfigMaps that contain PerformanceProfiles and Tuned? In the NodePool controller we can use type assertion to figure out which it is. The benefit being that it keeps the NodePool API simpler.

Reply:
Yeah I totally agree with that, in fact I was just proposing to use a different field just in case you would prefer to avoid mixing PAO and NTO configs.